### PR TITLE
add: verification of web3 user email on server (#59)

### LINKED ISSUE #59: Server should verify new user email from W3
When user signs up from W3, we pull details from W3 on the client.
This is a failure point since the client can to the server any information.
Server must verify the email the user sent is really from our W3 backend

Two options:
- W3 signs user data with timestamp valid for 1 minute and client send this signature to backend with his signup details
- OR client send w3 token with signup details and server request user details

### PR DESCRIPTION
**Pull request by yaroslav-fedyshyn-nordwhale**
This Pr closes #59 
 * make api route
 * send request to web3 site with web3 token
 * compare email from dapp and from web3 user

### CODE REVIEW

File: src/server/verification/verificationAPI.js [hunk 1]
```diff
@@ -282,6 +283,63 @@ const setup = (app: Router, verifier: VerificationAPI, storage: StorageAPI) => {
...
+      let _w3User
+
+      try {
+        _w3User = await crossFetch(`${conf.web3SiteUrl}/api/wl/user`, {
+          method: 'GET',
+          headers: {
+            Authorization: token
+          }
+        }).then(res => res.json())
+      } catch (e) {}
```
Comment (sirpy) [inline src/server/verification/verificationAPI.js]: log errors

File: src/server/verification/verificationAPI.js [hunk 2]
```diff
@@ -282,6 +283,63 @@ const setup = (app: Router, verifier: VerificationAPI, storage: StorageAPI) => {
...
+        message: 'Invalid web3 token'
+      }
+
+      if (_w3User) {
+        const w3User = _w3User.data
+
+        if (w3User.email === email) {
+          responsePayload.ok = 1
+          delete responsePayload.message
+
```
Comment (sirpy) [inline src/server/verification/verificationAPI.js]: here you need to update the user record
see: verify/email endpoint

**Comment by yaroslav-fedyshyn-nordwhale** [comment]
> Where is the PR for the client side?

Changes for client side is here in this PR - https://github.com/GoodDollar/GoodDAPP/pull/478

File: src/server/verification/verificationAPI.js (hunk 1 above)
Comment (sirpy) [reply src/server/verification/verificationAPI.js]: @yaroslav-fedyshyn-nordwhale

Comment (sirpy) [reply src/server/verification/verificationAPI.js]: we use log.error for errors

**Comment by sirpy** [comment]
use log.error for errors

**Comment by sirpy** [comment]
@yaroslav-fedyshyn-nordwhale resolve conflicts

**Pull request merged**